LAPACK should be a required dependency

Observed by @AGonzalezNicolas in https://github.com/HPSCTerrSys/TSMP2/issues/113#event-21237732897. Missing LAPACK triggers the ff. linker errors:

```bash
[100%] Linking Fortran executable eclm.exe
ld: ../clm5/libclm.a(SoilWaterMovementMod.F90.o): in function `dgtsv_.t253890p.t253891p.t253892p.t253893p.t253894p.t253895p.t253896p.t253897p':
eCLM/src/clm5/biogeophys/SoilWaterMovementMod.F90:(.text+0x280855): undefined reference to `dgtsv_'
ld: ../clm5/libclm.a(UrbBuildTempOleson2015Mod.F90.o): in function `dgesv_.t58174p.t58175p.t58176p.t58177p.t58178p.t58179p.t58180p.t58181p':
eCLM/src/clm5/biogeophys/UrbBuildTempOleson2015Mod.F90:(.text+0x237ae5): undefined reference to `dgesv_'
ld: ../clm5/libclm.a(BandDiagonalMod.F90.o): in function `dgbsv_.t839p.t840p.t841p.t842p.t843p.t844p.t845p.t846p.t847p.t848p':
eCLM/src/clm5/biogeophys/BandDiagonalMod.F90:(.text+0x28eaf): undefined reference to `dgbsv_'
```

The absence of LAPACK should be caught at the CMake configure stage.

https://github.com/HPSCTerrSys/eCLM/blob/91d5f7b4036a9e117c5ea9f999d7663f68fe6b90/src/clm5/CMakeLists.txt#L322-L325

--- a/src/clm5/CMakeLists.txt
+++ b/src/clm5/CMakeLists.txt
@@ -319,7 +319,7 @@ target_sources(${PROJECT_NAME}
         utils/spmdMod.F90
 )
 
-find_package(LAPACK)
+find_package(LAPACK REQUIRED)
 if(LAPACK_FOUND)
     target_link_libraries(${PROJECT_NAME} PRIVATE LAPACK::LAPACK)
 endif()

